File: 3June2024/Ques4.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<vector<int>> ans;

void solve(vector<int>& arr, int target, int n, vector<int>& temp) {
    if (target == 0) {
        ans.push_back(temp);
        return;
    }
    if (n < 0 || target < 0) return;
    temp.push_back(arr[n]);
    solve(arr, target - arr[n], n - 1, temp);
    temp.pop_back(); 
    solve(arr, target, n - 1, temp);
}

int main() {
    vector<int> arr{1, 2, 3, 4, 5, 8};
    int target = 9;
    vector<int> temp;

    solve(arr, target, arr.size() - 1, temp);

    for (const auto& subset : ans) {
        for (int num : subset) {
            cout << num << " ";
        }
        cout << endl;
    }

    return 0;
}

File: 6June2024/Ques4.java
public class Ques4 {
    public static void main(String[] args) {
        int arr[]={0,5, 8, 3, 5, 9, 2, 8, 6, 3, 1, 4, 7, 6, 2, 9};
        int temp[]=new int[arr.length];
        int ans=arr.length+1;
        for(int i=1;i<=arr.length;i++)
        {
            if(temp[arr[i-1]]==0)
            {
                temp[arr[i-1]]=i;
            }
            else
            {
                if(temp[arr[i-1]]<ans) ans=temp[arr[i-1]];
            }
        }
        System.out.println(ans);

    }
}
